src/measure_theory/lebesgue_measure.lean

```diff
@@ -379,6 +379,114 @@ begin
+  { exact measurable_set_region_between hf hg hs },
+end
+
+/-- The volume of the region between two almost everywhere measurable functions on a measurable set
+    can be represented as a Lebesgue integral. -/
+theorem volume_region_between_eq_lintegral [sigma_finite μ]
+  (hf : ae_measurable f (μ.restrict s)) (hg : ae_measurable g (μ.restrict s))
+  (hs : measurable_set s) :
+  μ.prod volume (region_between f g s) = ∫⁻ y in s, ennreal.of_real ((g - f) y) ∂μ :=
+begin
+  have h₁ : (λ y, ennreal.of_real ((g - f) y))
+          =ᵐ[μ.restrict s]
+              λ y, ennreal.of_real ((ae_measurable.mk g hg - ae_measurable.mk f hf) y) :=
+    eventually_eq.fun_comp (eventually_eq.sub hg.ae_eq_mk hf.ae_eq_mk) _,
+  have h₂ : (μ.restrict s).prod volume (region_between f g s) =
+    (μ.restrict s).prod volume (region_between (ae_measurable.mk f hf) (ae_measurable.mk g hg) s),
+  { apply measure_congr,
+    apply eventually_eq.inter, { refl },
+    exact eventually_eq.inter
+            (eventually_eq.comp₂ (ae_eq_comp' measurable_fst hf.ae_eq_mk
+              measure.prod_fst_absolutely_continuous) _ eventually_eq.rfl)
+            (eventually_eq.comp₂ eventually_eq.rfl _
+              (ae_eq_comp' measurable_fst hg.ae_eq_mk measure.prod_fst_absolutely_continuous)) },
+  rw [lintegral_congr_ae h₁,
+      ← volume_region_between_eq_lintegral' hf.measurable_mk hg.measurable_mk hs],
+  convert h₂ using 1,
+  { rw measure.restrict_prod_eq_prod_univ,
+    exacts [ (measure.restrict_eq_self_of_subset_of_measurable (hs.prod measurable_set.univ)
+      (region_between_subset f g s)).symm, hs], },
+  { rw measure.restrict_prod_eq_prod_univ,
+    exacts [(measure.restrict_eq_self_of_subset_of_measurable (hs.prod measurable_set.univ)
+      (region_between_subset (ae_measurable.mk f hf) (ae_measurable.mk g hg) s)).symm, hs] },
+end
+
+
+theorem volume_region_between_eq_integral' [sigma_finite μ]
+  (f_int : integrable_on f s μ) (g_int : integrable_on g s μ)
+  (hs : measurable_set s) (hfg : f ≤ᵐ[μ.restrict s] g ) :
+  μ.prod volume (region_between f g s) = ennreal.of_real (∫ y in s, (g - f) y ∂μ) :=
+begin
+  have h : g - f =ᵐ[μ.restrict s] (λ x, nnreal.of_real (g x - f x)),
+  { apply hfg.mono,
+    simp only [nnreal.of_real, max, sub_nonneg, pi.sub_apply],
+    intros x hx,
+    split_ifs,
+    refl },
+  rw [volume_region_between_eq_lintegral f_int.ae_measurable g_int.ae_measurable hs,
+    integral_congr_ae h, lintegral_congr_ae,
+    lintegral_coe_eq_integral _ ((integrable_congr h).mp (g_int.sub f_int))],
+  simpa only,
+end
+
+/-- If two functions are integrable on a measurable set, and one function is less than
+    or equal to the other on that set, then the volume of the region
+    between the two functions can be represented as an integral. -/
+theorem volume_region_between_eq_integral [sigma_finite μ]
+  (f_int : integrable_on f s μ) (g_int : integrable_on g s μ)
+  (hs : measurable_set s) (hfg : ∀ x ∈ s, f x ≤ g x) :
```

Comment:
Can you also weaken the assumption `hfg` here, replacing it with an almost everywhere assumption?

Reply:
@sgouezel Did you see that Andrew provided this as a variant just above, `volume_region_between_eq_integral'`?

Reply:
Over Zulip PMs, @hrmacbeth suggested that I commit two separate theorems for the sake of convenience: `volume_region_between_eq_integral'` (where `hfg` is an almost everywhere assumption) and `volume_region_between_eq_integral` (where `hfg` is the stronger assumption here). 

Are you suggesting that we don't need the version with a stronger assumption? Just making sure I understand your comment.

Reply:
Oops, my bad :-)